app/controllers/organisations_controller.rb

```diff
@@ -25,25 +25,25 @@ def show
   private
 
   def prepare_courses
-    cohorts = @organisation.cohorts.includes(:course).active.uniq
+    cohorts = @organisation.cohorts.includes(:course).uniq
 
     courses =
       cohorts.each_with_object({}) do |cohort, courses|
         courses[cohort.course.id] ||= { course: cohort.course, cohorts: [] }
-        courses[cohort.course.id][:cohorts] << cohort
+        if cohort.ends_at.nil? || cohort.ends_at.future?
+          courses[cohort.course.id][:cohorts] << cohort
+        end
       end
 
     courses.values.map do |course|
       cohort_ids = course[:cohorts].map(&:id)
 
-      course[:total_students] =
-        @organisation
-          .users
-          .joins(:founders)
-          .where(founders: { cohort_id: cohort_ids })
-          .distinct
-          .count
-
+      course[:total_students] = @organisation
+        .users
+        .joins(:founders)
+        .where(founders: { cohort_id: cohort_ids })
+        .distinct
+        .count
       course
     end
```

Comment:
I think it would be nice to:

1. Show information about _ended_ cohorts along with active number.
2. Show the _View All Cohorts_ link only when there is an actual need for it.

Something like...

```ruby
courses.values.map do |course|
  cohort_ids = course[:cohorts].map(&:id)
  inactive_cohort_ids = @organisation.cohorts.where.not(id: cohort_ids).pluck(:id)  
  course[:inactive_cohorts] = inactive_cohort_ids.any?
  course[:active_students] = student_count(cohort_ids)

  course[:inactive_students] = if course[:inactive_cohorts]
    student_count(course[:inactive_cohort_ids])
  else
    0
  end
end
```

...and then show this info in the UI, alongside the active student count.

Reply:
> Show information about ended cohorts along with active number.

Here we would like to show inactive cohorts count or inactive students count? Can you please elaborate? 

---

> Show the View All Cohorts link only when there is an actual need for it.

For above comment, 

- Show link to view all cohorts only when there are inactive cohorts
- Show link to view all cohorts only when there are active cohorts above a certain number. I think this case will not be required because   definitely for a course in an organisation there can can't be active cohorts more than 2, most probably in single number at a time. 

Do you see any other cases?

Reply:
@pupilfirst/designers 

> Show the View All Cohorts link only when there is an actual need for it.

What do you think about the above statement? 

Should we prefer consistency and clarity or efficiency and utility?

I felt the page lacked consistency after hiding `View All Cohorts` for a course with only active cohorts.

I also felt what Hari has mentioned is reasonable because we already show all active cohorts of a course on the organisation show page.

I am not sure what is good here. Please guide me